tensorflow/core/kernels/sparse_tensor_dense_matmul_op.cc

```diff
@@ -132,22 +147,39 @@ class SparseTensorDenseMatMulOp : public OpKernel {
       return;
     }
 
-#define MAYBE_ADJOINT(ADJ_A, ADJ_B)                                        \
-  if (adjoint_a_ == ADJ_A && adjoint_b_ == ADJ_B) {                        \
-    Status functor_status = functor::SparseTensorDenseMatMulFunctor<       \
-        Device, T, Tindices, ADJ_A,                                        \
-        ADJ_B>::Compute(ctx->eigen_device<Device>(), out->matrix<T>(),     \
-                        a_indices->matrix<Tindices>(), a_values->vec<T>(), \
-                        b->matrix<T>());                                   \
-    OP_REQUIRES_OK(ctx, functor_status);                                   \
+    switch (ndim) {
+#define MAYBE_ADJOINT(ADJ_A, ADJ_B, N)                               \
```

Comment:
An alternative approach would be to pass out and b reshaped to 3-tensors and N as a separate input.  there's a helper for this: flat_outer_dims:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.h#L305